crypto/s2n_hkdf.c

```diff
@@ -83,6 +85,29 @@ static int s2n_hkdf_expand(struct s2n_hmac_state *hmac, s2n_hmac_algorithm alg,
     return 0;
 }
 
+int s2n_hkdf_expand_label(struct s2n_hmac_state *hmac, s2n_hmac_algorithm alg, const struct s2n_blob *secret, const struct s2n_blob *label,
+                          const struct s2n_blob *context, struct s2n_blob *output)
+{
+    /* Per RFC8446: 7.1, a HKDF label is a 2 byte length field, and two 1...255 byte arrays with a one byte length field each. */
+    uint8_t hkdf_label_buf[2 + 256 + 256];
+    struct s2n_blob hkdf_label_blob = {0};
+    struct s2n_stuffer hkdf_label = {{0}};
+
+    GUARD(s2n_blob_init(&hkdf_label_blob, hkdf_label_buf, sizeof(hkdf_label_buf)));
+    GUARD(s2n_stuffer_init(&hkdf_label, &hkdf_label_blob));
+    GUARD(s2n_stuffer_write_uint16(&hkdf_label, output->size));
+    GUARD(s2n_stuffer_write_uint8(&hkdf_label, label->size + sizeof("tls13 ") - 1));
```

Comment:
Per the rfc:


   Note: With common hash functions, any label longer than 12 characters
   requires an additional iteration of the hash function to compute.
   The labels in this specification have all been chosen to fit within
   this limit.

Do we want to add a guard on that?

Reply:
Good idea! I've added an lte_check().

Reply:
looks good, its not clear on if the 12 bytes includes the `"tls13 "` prefix.  looking at the examples in the rfc, however indicates that `"traffic upd"` is a valid label and is 11 bytes long.  So cool, your check looks good :)